src/config.rs

```diff
@@ -506,7 +506,10 @@ fn config_from_env() -> Result<EnvConfig> {
     // ======= AWS =======
     let s3 = env::var("SCCACHE_BUCKET").ok().map(|bucket| {
         let region = env::var("SCCACHE_REGION").ok();
-        let no_credentials = env::var("SCCACHE_S3_NO_CREDENTIALS").ok().is_some();
+        let no_credentials = env::var("SCCACHE_S3_NO_CREDENTIALS")
```

Comment:
I think we should be more aggressive here.

if SCCACHE_S3_NO_CREDENTIALS is set, it can be only true or false (or 1 or 0).
Otherwise, we don't start the program.

Reply:
@sylvestre, I just pushed some changes. I'm not great with `rust`. Can you tell me if my approach is right? It took me a while to figure out the `Option` / `Result` handling.

I'll update the tests in a follow-up commit.